src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java

```diff
@@ -455,11 +455,14 @@ private static Connection getClosedConnection()
             public Object invoke(Object proxy, Method method, Object[] args) throws Throwable
             {
                final String methodName = method.getName();
-               if ("abort".equals(methodName)) { 
+               if ("isValid".equals(methodName)) {
+                  return Boolean.FALSE;
+               }
+               else if ("close".equals(methodName)) { 
```

Comment:
its common to unwrap and call close. let pool handle this case gracefully.
wrt checkException(), I have not made any change there in this pool request.